### ISSUE TITLE
Potential bug in getting prediction probs? 

### ISSUE THREAD

**Issue by TushaarGVS** [timothelaborie/text_classification_scripts#3]
In the unsloth text classification notebook, in considering:
```
par = torch.nn.Parameter(
    torch.vstack(
        [
            model.lm_head.weight[yes_token_id, :], 
            model.lm_head.weight[no_token_id, :]
         ]
    )
)
```

Wouldn't the `lm_head` logits at index-0 be for "Yes" token and index-1 be for "No" token: when you take an argmax over probabilities, you get a 1 for "No", and a 0 for "Yes"?

(Maybe I'm missing something?)

**Comment by timothelaborie** [comment]
good observation! I realize now that the tokens were mixed up. I made a new commit to fix this.
I also improved the prompt, now training is more stable and the accuracy went from 96.2 to 96.8!
